python/pyarrow/tests/test_extension_type.py

```diff
@@ -229,7 +238,7 @@ def test_ext_array_to_pylist():
     storage = pa.array([b"foo", b"bar", None], type=pa.binary(3))
     arr = pa.ExtensionArray.from_storage(ty, storage)
 
-    assert arr.to_pylist() == [b"foo", b"bar", None]
+    assert [x.as_py() if x else None for x in arr.to_pylist()] == [b"foo", b"bar", None]
```

Comment:
Hmm, don't we want the override to also affect `to_pylist`?

Reply:
Yeah. It seems that if `__arrow_ext_scalar_class__` is not defined on the ExtensionType we `.as_py()` is not available.

Reply:
This is resolved now.